app/src/ext_power_generic.c

```diff
@@ -27,6 +29,26 @@ struct ext_power_generic_data {
     bool status;
 };
 
+static bool ext_power_generic_state;
+static bool ext_power_generic_state_initalized = false;
+
+#if IS_ENABLED(CONFIG_SETTINGS)
+static void ext_power_save_state_work() {
+    settings_save_one("ext_power/state", &ext_power_generic_state, sizeof(ext_power_generic_state));
```

Comment:
Why not `sprintf(setting_name, "ext_power/state/%s", DT_INST_LABEL(0));` to future proof this for multiple?

Reply:
Thanks for this. It was the missing piece in how I would be able to generalize the solution more. I've updated to use this as well as cut the global state for the driver in favor of using status variables on the driver.